libs-scala/grpc-utils/src/main/scala/com/digitalasset/grpc/GrpcStatus.scala

```diff
@@ -3,21 +3,56 @@
 
 package com.daml.grpc
 
-import com.google.rpc.status.{Status => ProtobufStatus}
-import io.grpc.Status
+import com.google.protobuf.any.{Any => AnyProto}
+import com.google.protobuf.{Any => AnyJavaProto}
+import com.google.rpc.status.{Status => StatusProto}
+import com.google.rpc.{Status => StatusJavaProto}
 import io.grpc.Status.Code
+import io.grpc.{Metadata, Status}
+
+import scala.jdk.CollectionConverters._
 
 object GrpcStatus {
   type Description = Option[String]
+  type CodeValue = Int
 
   def unapply(status: Status): Some[(Code, Description)] =
     Some((status.getCode, Option(status.getDescription)))
 
-  def toProto(code: Code, description: Description): ProtobufStatus =
-    ProtobufStatus(code.value, description.getOrElse(""))
+  /** As [[io.grpc.Status]] and [[com.google.rpc.status.Status]] aren't isomorphic i.e. the former one
+    * doesn't contain details, this function takes an additional `metadata` argument to restore them.
+    */
+  def toProto(status: Status, metadata: Metadata): StatusProto = {
+    val code = status.getCode.value
+    val description = Option(status.getDescription).getOrElse("")
+    val statusJavaProto = io.grpc.protobuf.StatusProto.fromStatusAndTrailers(status, metadata)
+    val details =
+      statusJavaProto.getDetailsList.asScala.map(any => AnyProto.fromJavaProto(any)).toSeq
```

Comment:
```suggestion
      statusJavaProto.getDetailsList.asScala.map(AnyProto.fromJavaProto).toSeq
```

Reply:
Can't do that, as our linter fails on it. And `AnyProto.fromJavaProto(_)` gets a warning in the IDE :(